server/pogo/state.go

```diff
@@ -77,15 +77,38 @@ func (s *GameState) GetPlayer(id int) *PlayerState {
 	return nil
 }
 
+// GetCardType returns the type of the card with the matching id
+func (s *GameState) GetCardType(id int) CardType {
+	for _, player := range s.Players {
+		for _, card := range player.Hand {
+			if card.ID == id {
+				return card.CardType
+			}
+		}
+		for _, card := range player.Deck {
+			if card.ID == id {
+				return card.CardType
+			}
+		}
+		for _, card := range player.Discard {
+			if card.ID == id {
+				return card.CardType
+			}
+		}
+	}
+	return Unknown
+}
+
 // PlayerState contains all of the information about a given player, their hand, their discard, everything
 type PlayerState struct {
 	ID       int         // unique id for this player in this game
 	Name     string      // essentially the username of the player // TODO: MW is this necessary?
 	Hand     []CardState // the cards currently available to the player
 	Discard  []CardState // the cards that have been played or discarded
 	Deck     []CardState // the cards still in the deck
-	Location int         // the id of the board space this player is occupying
+	Location int         // the index of the board space the player is on
```

Comment:
Notice, the definition of Player.Location changed.  It used to be the ID of the space the player was standing on.  Now it is the index of that space.